spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -109,6 +144,90 @@ public void setTablePrefix(String tablePrefix) {
 		this.tablePrefix = tablePrefix;
 	}
 
+	@Override
+	public TaskExecution getTaskExecution(String executionId) {
+		try {
+			TaskExecution taskExecution = jdbcTemplate.queryForObject(getQuery(GET_EXECUTION_BY_ID),
+					new TaskExecutionRowMapper(), executionId);
+			taskExecution.setParameters(getTaskParameters(executionId));
+			return taskExecution;
+		}
+		catch (EmptyResultDataAccessException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public long getTaskExecutionCount(String taskName) {
+		try {
+		return jdbcTemplate.queryForObject(
+				getQuery(TASK_EXECUTION_COUNT), new Object[] { taskName }, Long.class);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return 0;
+		}
+
+	}
+
+	@Override
+	public Set<TaskExecution> findRunningTaskExecutions(String taskName) {
+		final Set<TaskExecution> result = new HashSet<TaskExecution>();
+		RowCallbackHandler handler = new RowCallbackHandler() {
+			@Override
+			public void processRow(ResultSet rs) throws SQLException {
+				TaskExecutionRowMapper mapper = new TaskExecutionRowMapper();
+				result.add(mapper.mapRow(rs, 0));
+			}
+		};
+		jdbcTemplate.query(getQuery(FIND_RUNNING_TASK_EXECUTIONS),
+				new Object[] { taskName }, handler);
```

Comment:
Why use a `RowCallbackHandler` here instead of just a `RowMapper` with `queryForObject`?